DisguiseUnityRenderStream/Editor/DisguiseRenderStreamBuild.cs

```diff
@@ -26,93 +49,76 @@ public void OnPreprocessBuild(BuildReport report)
         
-        /// <summary>
-        /// Ensure the proper runtime availability of the shader name.
-        /// </summary>
-        /// <remarks>
-        /// Based on logic exposed here:
-        /// https://forum.unity.com/threads/modify-always-included-shaders-with-pre-processor.509479/
-        /// </remarks>
-        /// <param name="shaderName">The name of the shader to validate.</param>
-        static void AddAlwaysIncludedShader(string shaderName)
+        string IUnityLinkerProcessor.GenerateAdditionalLinkXmlFile(BuildReport report, UnityLinkerBuildPipelineData data)
         {
-            var shader = Shader.Find(shaderName);
-            if (shader == null)
-                return;
-
-            var graphicsSettingsObj = AssetDatabase.LoadAssetAtPath<GraphicsSettings>("ProjectSettings/GraphicsSettings.asset");
-            var serializedObject = new SerializedObject(graphicsSettingsObj);
-            var arrayProp = serializedObject.FindProperty("m_AlwaysIncludedShaders");
-            var hasShader = false;
-            for (int i = 0; i < arrayProp.arraySize; ++i)
+            var preserver = new ReflectedMemberPreserver();
+            
+            GenerateSchema(report, scene =>
             {
-                var arrayElem = arrayProp.GetArrayElementAtIndex(i);
-                if (shader == arrayElem.objectReferenceValue)
+                var remoteParameters = Object.FindObjectsByType<DisguiseRemoteParameters>(FindObjectsSortMode.InstanceID);
+
+                foreach (var remoteParameter in remoteParameters)
                 {
-                    hasShader = true;
-                    break;
+                    foreach (var exposedParameter in remoteParameter.exposedParameters())
+                    {
+                        var memberInfo = remoteParameter.GetMemberInfoFromManagedParameter(exposedParameter);
+                        preserver.Preserve(memberInfo);
+                    }
                 }
-            }
+                
+                ProcessSceneForSchema(scene);
+            });
+            
+            m_HasGeneratedSchema = true;
 
-            if (!hasShader)
+            return preserver.GenerateAdditionalLinkXmlFile();
+        }
+        
+        void IPostprocessBuildWithReport.OnPostprocessBuild(BuildReport report)
+        {
+            if (m_NeededVSyncFix)
             {
-                var arrayIndex = arrayProp.arraySize;
-                arrayProp.InsertArrayElementAtIndex(arrayIndex);
-                var arrayElem = arrayProp.GetArrayElementAtIndex(arrayIndex);
-                arrayElem.objectReferenceValue = shader;
-
-                serializedObject.ApplyModifiedProperties();
-
-                AssetDatabase.SaveAssets();
+                Debug.LogWarning($"DisguiseRenderStream: {nameof(QualitySettings)}.{nameof(QualitySettings.vSyncCount)} is currently enabled in the project setting. Disabling vSync automatically for best output performance for Disguise.");
```

Comment:
Minor: Don't know if it is my understanding of English or its the phrase, but for me it sounds like if we don't change the setting but the setting will be ignore when we run.  When in fact we change the setting.  What about something like:

> DisguiseRenderStream: {nameof(QualitySettings)}.{nameof(QualitySettings.vSyncCount)} has been disabled for best output performance with Disguise.